go/vt/vtgate/evalengine/expr_compare.go

```diff
@@ -114,7 +114,7 @@ func (compareNullSafeEQ) compare(collationEnv *collations.Environment, left, rig
 }
 
 func typeIsTextual(tt sqltypes.Type) bool {
-	return sqltypes.IsTextOrBinary(tt) || tt == sqltypes.Time
+	return sqltypes.IsTextOrBinary(tt) || tt == sqltypes.Time || tt == sqltypes.Enum || tt == sqltypes.Set
```

Comment:
This is the follow-up fix for: https://github.com/vitessio/vitess/issues/15598

Reply:
@mattlord See https://github.com/vitessio/vitess/pull/15783 for the proper fix (which actually also includes this change, but it's only used as fallback / in case we don't compare two `enum` or `set` values :smile:.